hazelcast/src/main/java/com/hazelcast/map/impl/AbstractEvictableRecordStore.java

```diff
@@ -382,6 +383,12 @@ protected void accessRecord(Record record, long now) {
         setExpirationTime(record, maxIdleMillis);
     }
 
+    protected void mergeRecordExpiration(Record record, EntryView mergingEntry) {
+        record.setTtl(mergingEntry.getTtl());
```

Comment:
I think setting only TTL can still be problematic here. We should also replicate `mergingEntry.getLastUpdateTime()` for TTL based eviction and `mergingEntry.getLastAccessTime()` for idle eviction. By replicating those values, entries life spans will be much closer to the expected ones.

For example, by making this change; if merging entries TTL equals to 100 seconds and remaining life span is 10 seconds, we can expect that entry will be expired after 10 seconds later in replica cluster instead of 100 seconds.